Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,399 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Combine
+import Foundation
+
+extension RunValveIsolationTraceView {
+    /// The view model for this sample.
+    @MainActor
+    class Model: ObservableObject {
+        /// A map with a 'night time' street map style.
+        let map = Map(basemapStyle: .arcGISStreetsNight)
+        
+        /// The utility network for this sample.
+        private let utilityNetwork = UtilityNetwork(url: .featureServiceURL)
+        
+        /// The service geodatabase used to create the feature layer.
+        private let serviceGeodatabase = ServiceGeodatabase(url: .featureServiceURL)
+        
+        /// The service geodatabase feature layers.
+        private var layers: [FeatureLayer] = []
+        
+        /// The last element that was added to the list of filter barriers.
+        ///
+        /// When an element contains more than one terminal, the user should be presented with the
+        /// option to select a terminal. Keeping a reference to the last added element provides ease
+        /// of access to save the user's choice.
+        @Published private(set) var lastAddedElement: UtilityElement?
+        
+        /// The current tracing related activity.
+        @Published private(set) var tracingActivity: TracingActivity? = .loadingServiceGeodatabase
+        
+        /// The base trace parameters.
+        private let traceParameters = UtilityTraceParameters(traceType: .isolation, startingLocations: [])
+        
+        /// The point geometry of the starting location.
+        private(set) var startingLocationPoint: Point?
+        
+        /// The filter barrier categories.
+        private(set) var filterBarrierCategories: [UtilityCategory] = []
+        
+        /// The selected filter barrier category.
+        @Published private(set) var selectedCategory: UtilityCategory?
+        
+        /// A Boolean value indicating whether to include isolated features in the
+        /// trace results when used in conjunction with an isolation trace.
+        var includesIsolatedFeatures = true
```

Comment:
Instead of storing this on both the view and the model, either have it just on the model or, event better, pass it to the model as a parameter: `func trace(includeIsolatedFeatures:)`.
```

Reply:
I removed the property from the model and passed it as a parameter named `includesIsolatedFeatures` rather than `includeIsolatedFeatures` so that way the parameter matches the property name. Should the parameter name be `includeIsolatedFeatures` instead?

Reply:
I think `includeIsolatedFeatures` makes more sense as both the parameter name and as the state name.